scripts/mbedtls_framework/code_wrapper/psa_wrapper.py

```diff
@@ -0,0 +1,286 @@
+"""Generate wrapper functions for PSA function calls.
+"""
+
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
+
+import argparse
+import itertools
+import os
+from typing import Any, Iterator, List, Dict, Collection, Optional, Tuple
+
+from .. import build_tree
+from .. import c_parsing_helper
+from .. import c_wrapper_generator
+from .. import typing_util
+
+from .psa_buffer import BufferParameter
+
+class PSAWrapperConfiguration:
+    """Configuration data class for PSA Wrapper."""
+
+    cpp_guards = ["MBEDTLS_PSA_CRYPTO_C", "MBEDTLS_TEST_HOOKS", "!RECORD_PSA_STATUS_COVERAGE_LOG"]
```

Comment:
(Resuming https://github.com/Mbed-TLS/mbedtls-framework/pull/25#discussion_r1672417045)

> > These all should be instance variables, not class variables. Otherwise we will be very confused when we instantiate or override the class.
> >
> > Why isn't the class a [NamedTuple](https://docs.python.org/3/library/typing.html#typing.NamedTuple)?
>
> Can you elaborate on how that should be addressed?

Either use a `NamedTuple`, or make it a data class, as in, a class whose sole purpose is to store data into fields — what C would call a struct. Pylint doesn't like data classes, but it doesn't have a very strong justification. It wants to drive you to the mechanisms for building data classes through the standard library, but those mechanisms aren't particularly useful unless you want to do some introspection, which we don't care about here.

A custom data class would be just a constructor that initializes the fields to their default value.

```
class PSAWrapperConfiguration:
    """Configuration data class for PSA Wrapper."""
    def __init__(self) -> None:
        self.cpp_guards = ["MBEDTLS_PSA_CRYPTO_C", "MBEDTLS_TEST_HOOKS", "!RECORD_PSA_STATUS_COVERAGE_LOG"]
        …
```

Reply:
I am getting a bit of mixed signals, as to why we are going around in circles on this one.  The aim was to use a generic key:value configuration datastore, which allows other modules to import and .update() as required.

Then [as requested](https://github.com/Mbed-TLS/mbedtls-framework/pull/25#discussion_r1654996924) this was changed into a class because a using a dictionary is stringy typing? Now I am being asked to use a NamedTuple which completely fails the original intent, and is also stringy typing? 

Tuples are immutable, which means in order to update an entry the consumer will have to recreate it. There is a reason that dictionaries are universaly used for configuration input.

Now for dataclass argument, yes I understand that this is not a [dataclass](https://docs.python.org/3/library/dataclasses.html) as introduced in Python3.7, but the proposal is to basically implement in Python3.6 what the dataclass decorator does.

Is that all that for the shake of adding-complexity really needed?

Reply:
> The aim was to use a generic key:value configuration datastore, which allows other modules to import and .update() as required.

Uh? That's the first time this comes up. Why should this be a generic configuration datastore? Why would modules add their own configuration entries that the base class doesn't know about? That seems like a solution in need of a problem.

> a NamedTuple which completely fails the original intent, and is also stringy typing?

A NamedTuple isn't extensible. But extensibility is a new requirement and I don't see the point of this new requirement.

A NamedTuple is not stringly typed. Its type is the NamedTuple. The set of keys is known, and the type of the key corresponding to each value is known.

Note once again that I'm not saying you _have_ to use a NamedTuple. I'd be happy if you just made the change I posted above, changing the class fields into instance fields. With `# pylint: disable=too-few-public-methods` because Pylint is opinionated about not doing your own data class, but there isn't a strong objective justification behind that opinion.